trewrap/src/main/java/org/codice/imaging/nitf/trewrap/SECURA.java

```diff
@@ -0,0 +1,204 @@
+     *
+     * @return the security standard string
+     * @throws NitfFormatException if there is a parsing issue.
+     */
+    public final String getSecurityStandard() throws NitfFormatException {
+        return getValueAsTrimmedString("SECSTD");
+    }
+
+    /**
+     * Get the SECURITY field compression.
+     *
+     * This field identifies the compression used, if any, of the SECURITY field.
+     *
+     * @return the SECURITY field compression string
+     * @throws NitfFormatException if there is a parsing issue.
+     */
+    public final String getSecurityFieldCompression() throws NitfFormatException {
+        return getValueAsTrimmedString("SECCOMP");
+    }
+
+    /**
+     * Get the length of the SECURITY field.
+     *
+     * This field holds the length of the SECURITY field in bytes.
+     *
+     * @return SECURITY field length
+     * @throws NitfFormatException if there is a parsing issue on the NITF side
+     */
+    public final int getSecurityLength() throws NitfFormatException {
+        return getValueAsInteger("SECLEN");
+    }
+
+    /**
+     * Get the Security field value.
+     *
+     * The actual security data, as encoded using the security standard specified in the
+     * SECSTD field and compressed as specified by the SECCOMP field.
+     *
+     * @return the Security data
+     * @throws NitfFormatException if there is a parsing issue.
+     */
+    public final byte[] getSecurity() throws NitfFormatException {
+        return getFieldValue("SECURITY").getBytes(StandardCharsets.ISO_8859_1);
+    }
+
+    /**
+     * Get the Security field value.
+     *
+     * The actual security data, as encoded using the security standard specified in the
+     * SECSTD field and compressed as specified by the SECCOMP field.
+     *
+     * @return the uncompressed Security data
+     * @throws NitfFormatException if there is a parsing issue.
+     */
+    public final String getSecurityUncompressed() throws NitfFormatException {
+        if (getSecurityFieldCompression().equals("GZIP")) {
+            byte[] security = getSecurity();
+
+            StringBuilder sb = new StringBuilder();
+            try (ByteArrayInputStream bis = new ByteArrayInputStream(security);
+            GZIPInputStream gis = new GZIPInputStream(bis);
+            BufferedReader br = new BufferedReader(new InputStreamReader(gis, StandardCharsets.ISO_8859_1))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                sb.append(line);
+            }
+            } catch (IOException e) {
+                throw new NitfFormatException("Could not uncompress SECURITY field." + e.getLocalizedMessage());
+            }
+            return sb.toString();
+        }
+        return getFieldValue("SECURITY");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public final ValidityResult getValidity() throws NitfFormatException {
+        // TODO: add more validity checks
```

Comment:
Do you intend to implement this as part of this commit, or does it need a ticket to flag the future work?

Reply:
I'm not sure what's supposed to go in here, The wrapper I was using as a template only had this in it as well.  I'm ok with a ticket to flag it for future work.

Reply:
Added validity results.